Issutracker : #i46879#
CWS         : N/A
Author      : NAKATA Maho <maho@openoffice.org> (JCA)
Description : Need to include <sys/soundcard.h> to use sound and passes
              correct pthread related flags

Index: portaudio/portaudio_v18_1.patch
===================================================================
RCS file: /cvs/external/portaudio/portaudio_v18_1.patch,v
retrieving revision 1.1
diff -u -r1.1 portaudio_v18_1.patch
--- portaudio/portaudio_v18_1.patch	6 Oct 2004 11:48:04 -0000	1.1
+++ portaudio/portaudio_v18_1.patch	13 May 2005 23:27:21 -0000
@@ -12,7 +12,7 @@
   	;;
   
     mingw* )
---- 1883,1902 ----
+--- 1883,1921 ----
   case "${host_os}" in
     darwin* )
   
@@ -30,6 +30,25 @@
 ! 	PADLL="libportaudio.so";
 ! 	SHARED_FLAGS="-shared -Wl,-soname=libportaudio.so.0";
 ! 	CFLAGS="-O2 -fPIC"
+! 	;;
+! 
+!   freebsd* )
+! 	if test "$OSVERSION" -lt "500016"; then
+!              PTHREAD_CFLAGS="-D_THREAD_SAFE"
+!              PTHREAD_LIBS="-pthread"
+!         elif test "$OSVERSION" -lt "502102"; then
+!              PTHREAD_CFLAGS="-D_THREAD_SAFE"
+!              PTHREAD_LIBS="-lc_r"
+!         else
+!              PTHREAD_CFLAGS=""
+!              PTHREAD_LIBS="-pthread"
+!         fi
+! 
+! 	OTHER_OBJS="pa_unix_oss/pa_unix_oss.o pa_unix_oss/pa_unix.o";
+! 	LIBS="-lm ${PTHREAD_LIBS}";
+! 	PADLL="libportaudio.so";
+! 	SHARED_FLAGS="-shared -Wl,-soname=libportaudio.so.0";
+! 	CFLAGS="-O2 -fPIC ${PTHREAD_CFLAGS}"
   	;;
   
     mingw* )
@@ -52,3 +71,18 @@
   LIBINST = /usr/local/lib
   
   TESTS:= $(wildcard pa_tests/pa*.c pa_tests/debug*.c)
+
+*** misc/portaudio_v18_1/pa_unix_oss/pa_unix_oss.c	Tue Jul  1 00:05:50 2003
+--- misc/build/portaudio_v18_1/pa_unix_oss/pa_unix_oss.c	Sat May 14 08:14:16 2005
+***************
+*** 43,48 ****
+--- 43,50 ----
+  
+  #ifdef __linux__
+  #include <linux/soundcard.h>
++ #elif defined (__FreeBSD__)
++ #include <sys/soundcard.h>
+  #else
+  #include <machine/soundcard.h> /* JH20010905 */
+  #endif
+
